Automattic/wp-calypso
# Integrate new backup cards in Backup section

### Changes proposed in this Pull Request

This PR adds a new backup card and updates the successful backup screen in Calypso and Jetpack cloud, according to the new iteration of this screen. See mockups labelled `i4` (link is in the board).

Note that it doesn't modify the activity log in Jetpack cloud, which is what you see when you click the search icon in the _Backup_ section in cloud.

Fixes 1197351014149633-as-1197888024354846

### Implementation notes

- Extracted `Activity` types in own file
- Extracted enable restores banner in its own file
- Fixed some TS import errors by prepending paths with `calypso`
- Made `ActivityActor` support multiple avatar sizes and added the possibility to hide the actor info
- Refactored the structure of the Backup page for the new iteration
- Created an alternate component for `DailyBackupStatus` (named `DailyBackupStatusAlternate`). It was easier to decide which one to instanciate in `BackupsPage` rather than adding a ton of conditions in `DailyBackupStatus`.

_TODO: update feature flag name_

### Testing instructions

First part is to do a regression testing and check that the Backup section is behaving as it is in production. Then:

- Visit `/backup/:site/?flags=jetpack/backup-simplified-screens-i4`
- Check that the successful screen looks like the mockups
- Check that the updates make sense, and that no critical information has been lost with this new iteration
- Check that events are tracked when clicking on the restore and download buttons (respectively `calypso_jetpack_backup_restore`, and `calypso_jetpack_backup_download`)

A couple of cases/environments to test:
- Calypso vs. Jetpack cloud
- Desktop vs. mobile
- With and without server credentials
- Backup Daily vs Backup Real-time

### Screenshots

#### Calypso
_Daily backup with changes (before)_
<img width="775" alt="Screen Shot 2020-10-09 at 4 44 46 PM" src="https://user-images.githubusercontent.com/1620183/95638431-9bafe200-0a62-11eb-9c64-8c45bcfa32ec.png">

_Daily backup with changes (after)_
<img width="763" alt="Screen Shot 2020-10-09 at 4 45 00 PM" src="https://user-images.githubusercontent.com/1620183/95638447-a5394a00-0a62-11eb-8f0b-3999be81c6ed.png">

_Daily backup without changes (before)_
<img width="757" alt="Screen Shot 2020-10-09 at 4 45 57 PM" src="https://user-images.githubusercontent.com/1620183/95638498-d6197f00-0a62-11eb-8ac1-68b344288e61.png">

_Daily backup without changes (after)_
<img width="759" alt="Screen Shot 2020-10-09 at 4 46 01 PM" src="https://user-images.githubusercontent.com/1620183/95638505-de71ba00-0a62-11eb-8f39-310fd3a7df96.png">
